lights-out.js

```diff
@@ -16,4 +17,51 @@ class LightsOut {
             }
         }
     }
+
+    getNeighbors(row, col) {
+        const neighborArray = [];
+        
+        if(row < this.numRows - 1) {
+            if(row > 0) {
+                neighborArray.push([row + 1, col]);
+                neighborArray.push([row - 1, col]);
+            } else {
+                neighborArray.push([row + 1, col]);
+                neighborArray.push([this.numRows - 1, col]);
+            }
+        } else {
+            neighborArray.push([0, col]);
+            neighborArray.push([row - 1, col]);
+        }
+        if(col < this.numCols - 1) {
+            if(col > 0) {
+                neighborArray.push([row, col + 1]);
+                neighborArray.push([row, col - 1]);
+            } else {
+                neighborArray.push([row, col + 1]);
+                neighborArray.push([row, this.numCols - 1]);     
+            }
+        } else {
+            neighborArray.push([row, 0]);
+            neighborArray.push([row, col - 1]);
+        }
+
+        console.log(neighborArray);
+        return neighborArray;
+    }
+
+    registerClicks() {
+        for(let row = 0; row < this.numRows; row++) {
+            for(let col = 0; col < this.numCols; col++) {
+                const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
+                button.addEventListener('click', () => {
+                    button.classList.toggle("on");
+                    const neighbors = this.getNeighbors(row, col);
+                    
+                    neighbors.forEach(neighbor => document.querySelector(`#${this.gameId} [data-row="${neighbor[0]}"][data-col="${neighbor[1]}"]`)

```

Comment:
I think the use of `forEach` in this context is a little confusing. I suggest instead:

```
for (const neighbor of neighbors) {
    const [r, c] = neighbor;
    document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
        .classList.toggle("on")
}
```

Reply:
Added `for (const neighbor of neighbors) {
    const [r, c] = neighbor;
    document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
        .classList.toggle("on")
}` to `registerClicks` and removed `neighbors.forEach(neighbor => document.querySelector(`#${this.gameId} [data-row="${neighbor[0]}"][data-col="${neighbor[1]}"]`)`